flutter/lib/src/sentry_screenshot_options.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:async';
+
+import '../sentry_flutter.dart';
+
+/// Configuration of the screenshot feature.
+class SentryScreenshotOptions {
+  /// Automatically attaches a screenshot when capturing an error or exception.
+  ///
+  /// Requires adding the [SentryWidget] to the widget tree.
+  /// Example:
+  /// runApp(SentryWidget(child: App()));
+  /// The [SentryWidget] has to be the root widget of the app.
+  bool attachScreenshot = false;
+
+  /// Sets a callback which is executed before capturing screenshots. Only
+  /// relevant if `attachScreenshot` is set to true. When false is returned
+  /// from the function, no screenshot will be attached.
+  BeforeScreenshotCallback? beforeScreenshot;
+
+  /// Only attach a screenshot when the app is resumed.
+  bool attachScreenshotOnlyWhenResumed = false;
+
+  /// The quality of the attached screenshot
+  SentryScreenshotQuality screenshotQuality = SentryScreenshotQuality.high;
+}
```

Comment:
This is "namespaced" under `options.screenshot` so we don't need to repeat "screenshot" in field names anymore. Maybe:

```suggestion
class SentryScreenshotOptions {
  /// Automatically attaches a screenshot when capturing an error or exception.
  ///
  /// Requires adding the [SentryWidget] to the widget tree.
  /// Example:
  /// runApp(SentryWidget(child: App()));
  /// The [SentryWidget] has to be the root widget of the app.
  bool attach = false;

  /// Sets a callback which is executed before capturing screenshots. Only
  /// relevant if `attachScreenshot` is set to true. When false is returned
  /// from the function, no screenshot will be attached.
  BeforeScreenshotCallback? beforeCapture;

  /// Only attach a screenshot when the app is resumed.
  bool attachOnlyWhenResumed = false;

  /// The quality of the attached screenshot
  SentryScreenshotQuality quality = SentryScreenshotQuality.high;
}
```

also, `  /// Only attach a screenshot when the app is resumed.` makes no sense to me. Do you know what it does? If so can you maybe update the comment, even though it's. unrelated to this PR?

Reply:
@vaind I found this in the docs: [Screenshot Integration Background Crash](https://docs.sentry.io/platforms/flutter/troubleshooting/#screenshot-integration-background-crash)
So I think we should leave this setting there. WDYT?

Reply:
well, another thing to keep track of in future Flutter version bumps then.